cmd/subnetcmd/import_from_api.go

```diff
@@ -204,72 +205,79 @@ func importRunningSubnet(cmd *cobra.Command, args []string) error {
 		vmType = models.VMTypeFromString(subnetTypeStr)
 	}
 
+	vmIDstr := vmID.String()
+
 	sc := &models.Sidecar{
-		Name:            subnetName,
-		VM:              vmType,
-		ImportedFromAPM: false,
+		Name: subnetName,
+		VM:   vmType,
 		Networks: map[string]models.NetworkData{
 			network.String(): {
 				SubnetID:     subnetID,
 				BlockchainID: blockchainID,
 			},
 		},
-		Subnet:  subnetName,
-		Version: constants.SidecarVersion,
+		Subnet:       subnetName,
+		Version:      constants.SidecarVersion,
+		TokenName:    constants.DefaultTokenName,
+		ImportedVMID: vmIDstr,
+		// signals that the VMID wasn't derived from the subnet name but through import
+		ImportedFromAPM: true,
 	}
 
 	var versions []string
-	switch vmType {
-	case models.SubnetEvm:
-		versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SubnetEVMRepoName)
-		if err != nil {
-			return err
+
+	if reply != nil {
+		// a node was queried
+		for _, v := range reply.VMVersions {
+			if v == vmIDstr {
+				sc.VMVersion = v
+				break
+			}
+		}
+		sc.RPCVersion = int(reply.RPCProtocolVersion)
+	} else {
+		// no node was queried, ask the user
+		switch vmType {
+		case models.SubnetEvm:
+			versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SubnetEVMRepoName)
+			if err != nil {
+				return err
+			}
+			sc.VMVersion, err = app.Prompt.CaptureList("Pick the version for this VM", versions)
+		case models.SpacesVM:
+			versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SpacesVMRepoName)
+			if err != nil {
+				return err
+			}
+			sc.VMVersion, err = app.Prompt.CaptureList("Pick the version for this VM", versions)
+		case models.CustomVM:
+			return fmt.Errorf("importing custom VMs is not yet implemented, but will be available soon")
+		default:
+			return fmt.Errorf("unexpected VM type: %v", vmType)
 		}
-		sc.VMVersion, err = app.Prompt.CaptureList("Pick the version for this VM", versions)
-	case models.SpacesVM:
-		versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SpacesVMRepoName)
 		if err != nil {
 			return err
 		}
-		sc.VMVersion, err = app.Prompt.CaptureList("Pick the version for this VM", versions)
-	case models.CustomVM:
-		return fmt.Errorf("importing custom VMs is not yet implemented, but will be available soon")
-	default:
-		return fmt.Errorf("unexpected VM type: %v", vmType)
-	}
-	if err != nil {
-		return err
 	}
 
-	// hasn't been set in reply
-	if sc.RPCVersion == 0 {
-		sc.RPCVersion, err = vm.GetRPCProtocolVersion(app, vmType, sc.VMVersion)
-		if err != nil {
-			return fmt.Errorf("failed getting RPCVersion for VM type %s with version %s", vmType, sc.VMVersion)
-		}
+	sc.RPCVersion, err = vm.GetRPCProtocolVersion(app, vmType, sc.VMVersion)
```

Comment:
Does this need to be inside the else? It looks like it's set in the `if`

Reply:
Great catch! I think so too.